py/_io/terminalwriter.py

```diff
@@ -255,6 +255,19 @@ def write(self, msg, **kw):
                 markupmsg = msg
             write_out(self._file, markupmsg)
 
+    def write_source(self, source):
+        """Writes the given source code highlighted"""
+        if self.hasmarkup:
+            try:
+                from pygments.formatters.terminal import TerminalFormatter
```

Comment:
There's also 256-color and true color support.. https://github.com/pygments/pygments/blob/master/pygments/formatters/terminal256.py

Something like this could be used for auto-detect, but it should be made configurable probably.

```
        pygments_formatter_class = "pygments.formatters.TerminalTrueColorFormatter"
        pygments_formatter_kwargs = {"bg": "light", "style": "solarized-light"}
```
(from my pdbpp config)